cicecore/cicedynB/dynamics/ice_dyn_evp.F90

```diff
@@ -341,35 +341,66 @@ subroutine evp (dt)
-                         uvel_init (:,:,iblk), vvel_init (:,:,iblk), &
-                         uvel      (:,:,iblk), vvel      (:,:,iblk), &
-                         Tbu       (:,:,iblk))
+         
+            if (grid_system == trim('B')) then
+            call dyn_prep2 (nx_block,             ny_block,             & 
+                            ilo, ihi,             jlo, jhi,             &
+                            icellt(iblk),         icellu(iblk),         & 
+                            indxti      (:,iblk), indxtj      (:,iblk), & 
+                            indxui      (:,iblk), indxuj      (:,iblk), & 
+                            aiu       (:,:,iblk), umass     (:,:,iblk), & 
+                            umassdti  (:,:,iblk), fcor_blk  (:,:,iblk), & 
+                            umask     (:,:,iblk),                       & 
+                            uocn      (:,:,iblk), vocn      (:,:,iblk), & 
+                            strairx   (:,:,iblk), strairy   (:,:,iblk), & 
+                            ss_tltx   (:,:,iblk), ss_tlty   (:,:,iblk), &  
+                            icetmask  (:,:,iblk), iceumask  (:,:,iblk), & 
+                            fm        (:,:,iblk), dt,                   & 
+                            strtltx   (:,:,iblk), strtlty   (:,:,iblk), & 
+                            strocnx   (:,:,iblk), strocny   (:,:,iblk), & 
+                            strintx   (:,:,iblk), strinty   (:,:,iblk), & 
+                            taubx     (:,:,iblk), tauby     (:,:,iblk), & 
+                            waterx    (:,:,iblk), watery    (:,:,iblk), & 
+                            forcex    (:,:,iblk), forcey    (:,:,iblk), & 
+                            stressp_1 (:,:,iblk), stressp_2 (:,:,iblk), & 
+                            stressp_3 (:,:,iblk), stressp_4 (:,:,iblk), & 
+                            stressm_1 (:,:,iblk), stressm_2 (:,:,iblk), & 
+                            stressm_3 (:,:,iblk), stressm_4 (:,:,iblk), & 
+                            stress12_1(:,:,iblk), stress12_2(:,:,iblk), & 
+                            stress12_3(:,:,iblk), stress12_4(:,:,iblk), & 
+                            uvel_init (:,:,iblk), vvel_init (:,:,iblk), &
+                            uvel      (:,:,iblk), vvel      (:,:,iblk), &
+                            Tbu       (:,:,iblk))
+            else  
+            call dyn_prep2 (nx_block,             ny_block,             &
+                            ilo, ihi,             jlo, jhi,             &
+                            icellt(iblk),         icellu(iblk),         &
+                            indxti      (:,iblk), indxtj      (:,iblk), &
+                            indxui      (:,iblk), indxuj      (:,iblk), &
+                            aiu       (:,:,iblk), umass     (:,:,iblk), &
+                            umassdti  (:,:,iblk), fcor_blk  (:,:,iblk), &
+                            umaskCD   (:,:,iblk),                       &
```

Comment:
It's unclear to me why we need yet another call to dyn_prep2 here. I thought that all CD grid quantities were being dealt with in the two calls to dyn_prep2 below (for 'E' and 'N' quantities, line 440 and 473 in this version). Here your are adding a new call, identical ( as far as I can tell) to the B-grid case, apart from this new `umaskCD` array instead of `umask`... But on the CD grid, we are not using  most of these arrays that are sent as arguments here...

Reply:
umaskCD is defined different, icellu will be different (stress_u). It can be that calling dyn_prep2 is too much and a call to a simpler function could be used.